ibis/omniscidb/tests/test_client.py

```diff
@@ -73,6 +73,16 @@ def test_database_layer(con, alltypes):
     assert db.list_tables() == con.list_tables()
 
 
+def test_separate_database(con, test_data_db, temp_database):
```

Comment:
how does this test the above?

this actually is a fine general test, can you move to the ibis/tests/all/test_client.py

Reply:
_Hmm, I thought I've answered, but it seems I didn't press the `Comment` button or something..._

I've added the comments explaining the test flow in the test, but I can re-iterate here:
1. The connection keeps an object of currently active database, and if you request a new database object with either `None` as name or with the name that equals current, it returns the existing object, but if you request other name it creates a new object.
2. This test goes as following - it requests a temporary database (this creates new connection with a newly made temp database as current), then it requests an existing database (which obviously isn't equal to current db name being a temp one), so the connection creates a new object
3. Assertions there aren't strictly needed for testing this particular fix, but a test without assertions looks strange :) Actually, if you run this test prior to my fix there would be `AttributeError` raised, so it's still testing stuff, but I decided to check a few things just to use those objects.

I can move it to the general test suite, sure.

Reply:
@jreback please take a look, we have moved the test and enabled it only for backends it has a meaning for.